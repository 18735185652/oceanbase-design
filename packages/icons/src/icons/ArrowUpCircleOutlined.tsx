// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import ArrowUpCircleOutlinedSvg from '../asn/ArrowUpCircleOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const ArrowUpCircleOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={ArrowUpCircleOutlinedSvg} />;

ArrowUpCircleOutlined.displayName = 'ArrowUpCircleOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(ArrowUpCircleOutlined);