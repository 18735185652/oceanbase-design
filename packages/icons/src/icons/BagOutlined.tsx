// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import BagOutlinedSvg from '../asn/BagOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const BagOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={BagOutlinedSvg} />;

BagOutlined.displayName = 'BagOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(BagOutlined);