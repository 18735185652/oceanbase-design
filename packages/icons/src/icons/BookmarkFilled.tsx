// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import BookmarkFilledSvg from '../asn/BookmarkFilled';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const BookmarkFilled = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={BookmarkFilledSvg} />;

BookmarkFilled.displayName = 'BookmarkFilled';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(BookmarkFilled);