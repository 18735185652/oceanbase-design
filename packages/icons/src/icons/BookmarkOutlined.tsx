// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import BookmarkOutlinedSvg from '../asn/BookmarkOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const BookmarkOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={BookmarkOutlinedSvg} />;

BookmarkOutlined.displayName = 'BookmarkOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(BookmarkOutlined);