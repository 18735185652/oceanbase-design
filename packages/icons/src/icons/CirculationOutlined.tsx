// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CirculationOutlinedSvg from '../asn/CirculationOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CirculationOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CirculationOutlinedSvg} />;

CirculationOutlined.displayName = 'CirculationOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CirculationOutlined);