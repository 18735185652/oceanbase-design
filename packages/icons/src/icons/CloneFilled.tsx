// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CloneFilledSvg from '../asn/CloneFilled';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CloneFilled = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CloneFilledSvg} />;

CloneFilled.displayName = 'CloneFilled';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CloneFilled);