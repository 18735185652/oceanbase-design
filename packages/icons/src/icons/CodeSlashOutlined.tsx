// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CodeSlashOutlinedSvg from '../asn/CodeSlashOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CodeSlashOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CodeSlashOutlinedSvg} />;

CodeSlashOutlined.displayName = 'CodeSlashOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CodeSlashOutlined);