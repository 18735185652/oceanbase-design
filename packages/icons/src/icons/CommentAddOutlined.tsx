// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CommentAddOutlinedSvg from '../asn/CommentAddOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CommentAddOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CommentAddOutlinedSvg} />;

CommentAddOutlined.displayName = 'CommentAddOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CommentAddOutlined);