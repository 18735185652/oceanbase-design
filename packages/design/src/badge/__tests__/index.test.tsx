import React from 'react';
import { render } from '@testing-library/react';
import { Badge } from '@oceanbase/design';
import { TaobaoOutlined } from '@ant-design/icons';

describe('Badge', () => {

  it('icon mode should render correctly', () => {
    const { container, asFragment } = render(<Badge status='success' icon={true} />);
    expect(container.querySelector('.ant-badge-status-icon')).toBeTruthy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('custom icon mode should render correctly', () => {
    const { container, asFragment } = render(
      <Badge status='success' icon={<TaobaoOutlined />} />
    );
    expect(container.querySelector('.custom-indicator')).toBeTruthy();
    expect(container.querySelector('.ant-spin-oceanbase')).toBeFalsy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
