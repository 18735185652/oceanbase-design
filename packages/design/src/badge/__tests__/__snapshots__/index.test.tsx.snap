// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge gray default indicator should render correctly 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center ant-badge"
>
  <div
    class="ant-space-item"
  >
    <span
      class="ant-badge-status-icon ant-badge-status-success"
    >
      <span
        aria-label="check-circle"
        class="anticon anticon-check-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="check-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;

exports[`Badge icon mode should render correctly 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center ant-badge"
>
  <div
    class="ant-space-item"
  >
    <span
      class="ant-badge-status-icon ant-badge-status-success"
    >
      <span
        aria-label="check-circle"
        class="anticon anticon-check-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="check-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;
